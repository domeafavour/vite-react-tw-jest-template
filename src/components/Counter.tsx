import { useState } from 'react';

export function useCounter(initialCount = 0) {
  const [count, setCount] = useState(initialCount);
  function increment() {
    setCount((c) => c + 1);
  }
  return [count, { increment }] as const;
}

export function Counter({ initialCount = 0 }: { initialCount?: number }) {
  const [count, { increment }] = useCounter(initialCount);
  return (
    <button type="button" onClick={increment}>
      {count}
    </button>
  );
}
