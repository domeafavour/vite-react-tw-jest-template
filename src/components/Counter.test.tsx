import { act, fireEvent, render, renderHook } from '@testing-library/react';
import { Counter, useCounter } from './Counter';

describe('useCounter', () => {
  it('default count is 0', () => {
    const { result } = renderHook(() => useCounter());
    expect(result.current[0]).toBe(0);
  });

  it('should return the value given', () => {
    const { result } = renderHook(() => useCounter(1));
    expect(result.current[0]).toBe(1);
  });

  it('should be 1 after increment', () => {
    const { result, rerender } = renderHook(() => useCounter());
    expect(result.current[0]).toBe(0);
    act(() => {
      result.current[1].increment();
    });
    expect(result.current[0]).toBe(1);
    rerender();
    expect(result.current[0]).toBe(1);
  });
});

describe('Counter', () => {
  it('should render without crash', () => {
    const result = render(<Counter />);
    expect(result.container).toMatchSnapshot();
  });

  it('should render without crash while `initialCount` is given', () => {
    const result = render(<Counter initialCount={1} />);
    expect(result.container).toMatchSnapshot();
  });

  it('should increment the count', () => {
    const result = render(<Counter />);
    expect(result.getByText('0')).toBeTruthy();
    act(() => {
      fireEvent.click(result.getByRole('button'));
    });
    expect(result.getByText('1')).toBeTruthy();
  });
});
