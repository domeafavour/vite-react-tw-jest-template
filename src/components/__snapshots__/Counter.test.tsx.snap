// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter should render without crash 1`] = `
<div>
  <button
    type="button"
  >
    0
  </button>
</div>
`;

exports[`Counter should render without crash while \`initialCount\` is given 1`] = `
<div>
  <button
    type="button"
  >
    1
  </button>
</div>
`;
